assets/component-complementary-products.css

```diff
@@ -107,17 +97,47 @@ product-recommendations.is-accordion + .accordion {
   margin-top: 0;
 }
 
+.complementary-products-contains-quick-add .underline-links-hover:hover a {
+  text-decoration: initial;
+}
+
+.complementary-products-contains-quick-add .card__heading:hover a {
+  text-decoration: underline;
+}
+
 .complementary-products__container .card--card .card__inner .card__media {
   border-radius: calc(var(--corner-radius) - var(--border-width) - var(--image-padding));
 }
 
 .complementary-products__container .card--horizontal .quick-add {
```

Comment:
I'd like to potentially do some cleanup at some point. Maybe it's something we will need to do if we reuse the horizontal card pattern anywhere else in the theme. 
Right some styling is too tied to `complementary` when it could be based only on the `card--horizontal` class in `component-card.css`.